testsuite/python/lb.py

```diff
@@ -106,17 +106,101 @@ def test_properties(self):
             self.lbf[0, 1].velocity = [1, 2, 3]
 
     def test_raise_if_not_active(self):
-        lbf = self.lb_class(visc=1.0, dens=1.0, agrid=1.0, tau=0.1)
+        class MockLBFluid(self.lb_class):
+            '''LB class mock that ignores runtime errors from agrid and tau.'''
+            @property
+            def agrid(self):
+                return 1.
+
+            @agrid.setter
+            def agrid(self, value):
+                pass
+
+            @property
+            def tau(self):
+                return 0.01
+
+            @tau.setter
+            def tau(self, value):
+                pass
+
+        self.check_raise_if_not_active(self.lb_class, False)
+        self.check_raise_if_not_active(MockLBFluid, True)
```

Comment:
Note: this mocked LB class was introduced to help testing exceptions thrown from core getters. This works around the function `assert_agrid_tau_set()` that intercepts some of the calls to core functions if agrid or tau is not set. Since this is a function from a compiled Cython file, it's not possible to redefine it from the Python interpreter. The mocked class solves the issue without too much code. In the walberla branch, we don't need this workaround.